LambdaEngine/Source/Resources/ResourceManager.cpp

```diff
@@ -368,22 +399,83 @@ namespace LambdaEngine
 		{
 			guid							= s_NextFreeGUID++;
 			ppMappedMaterial				= &s_Materials[guid]; //Creates new entry if not existing
+			s_MaterialGUIDsToNames[guid]	= name;
 			s_MaterialNamesToGUIDs[name]	= guid;
 		}
 
 		(*ppMappedMaterial) = DBG_NEW Material();
 
-		Texture* pAlbedoMap						= albedoMap				!= GUID_NONE ? s_Textures[albedoMap]				: s_Textures[GUID_TEXTURE_DEFAULT_COLOR_MAP];
-		Texture* pNormalMap						= normalMap				!= GUID_NONE ? s_Textures[normalMap]				: s_Textures[GUID_TEXTURE_DEFAULT_NORMAL_MAP];
-		Texture* pAmbientOcclusionMap			= ambientOcclusionMap	!= GUID_NONE ? s_Textures[ambientOcclusionMap]		: s_Textures[GUID_TEXTURE_DEFAULT_COLOR_MAP];
-		Texture* pMetallicMap					= metallicMap			!= GUID_NONE ? s_Textures[metallicMap]				: s_Textures[GUID_TEXTURE_DEFAULT_COLOR_MAP];
-		Texture* pRoughnessMap					= roughnessMap			!= GUID_NONE ? s_Textures[roughnessMap]				: s_Textures[GUID_TEXTURE_DEFAULT_COLOR_MAP];
+		Texture* pAlbedoMap;
+		Texture* pNormalMap;
+		Texture* pAmbientOcclusionMap;
+		Texture* pMetallicMap;
+		Texture* pRoughnessMap;
+
+		TextureView* pAlbedoMapView;
+		TextureView* pNormalMapView;
+		TextureView* pAmbientOcclusionMapView;
+		TextureView* pMetallicMapView;
+		TextureView* pRoughnessMapView;
 
-		TextureView* pAlbedoMapView				= albedoMap				!= GUID_NONE ? s_TextureViews[albedoMap]			: s_TextureViews[GUID_TEXTURE_DEFAULT_COLOR_MAP];
-		TextureView* pNormalMapView				= normalMap				!= GUID_NONE ? s_TextureViews[normalMap]			: s_TextureViews[GUID_TEXTURE_DEFAULT_NORMAL_MAP];
-		TextureView* pAmbientOcclusionMapView	= ambientOcclusionMap	!= GUID_NONE ? s_TextureViews[ambientOcclusionMap]	: s_TextureViews[GUID_TEXTURE_DEFAULT_COLOR_MAP];
-		TextureView* pMetallicMapView			= metallicMap			!= GUID_NONE ? s_TextureViews[metallicMap]			: s_TextureViews[GUID_TEXTURE_DEFAULT_COLOR_MAP];
-		TextureView* pRoughnessMapView			= roughnessMap			!= GUID_NONE ? s_TextureViews[roughnessMap]			: s_TextureViews[GUID_TEXTURE_DEFAULT_COLOR_MAP];
+		if (albedoMap != GUID_NONE)
```

Comment:
You can make a function or macro out of these if statements. The function/macro would probably accept a GUID and a default/fallback GUID.

Reply:
Good idea